### DISCUSSION TITLE
Realtime UPDATE not working

### DISCUSSION THREAD

**Discussion by piccinnigius** [supabase/supabase#16735] [Questions]
Hi,
my setup is:

- supabase local docker
- nextjs app (acting as client)
- python server (using service role)

the problem is that on the client i receive updates maded from the studio but not changes made from the python server.

EDIT:
- i have tried with and without RLS
- other tables works good
- on the server tried with upsert and update but no differences

despite enabling/disabling rls from the studio policies seems to still. then this is mine:

```sql
BEGIN;
  ALTER POLICY "Only uploader can see his recordings" ON "public"."recordings" USING ((uploaded_by = auth.uid()));
  ALTER POLICY "Only uploader can see his recordings" ON "public"."recordings" WITH CHECK ((uploaded_by = auth.uid()));
COMMIT;
```

if removed and replaced with simple authenticated check works good.

Any hint?
Thanks in advance

**Comment by GaryAustin1** [comment]
Is the python client actually making changes to the rows?  Seems like the RLS either lets it or not make the change.  Then realtime on the client does not care how or where the update came from.

**Reply by piccinnigius** [reply]
Hi @GaryAustin1 thanks for the answer! yes, i can see the updates throught the studio ui or by refreshing the page. The strange thing is that for insert and delete works as usual (i use the same callback)

**Reply by GaryAustin1** [reply]
If you are getting updates when changing from the table UI, then I have no idea.  I don't see how realtime would know who is doing the update.
Any filter on the realtime that might not get met with the updated data?

**Reply by piccinnigius** [reply]
nope, clean listen. this is the code:

```tsx
const onInsertOrUpdate = async (payload: any) => {
    console.log("onInsertOrUpdate", payload)
    if (payload.eventType === 'DELETE') {
      setRecordings((prev: any) => prev.filter((r: any) => r.id !== payload.old.id))
      return
    } else if (payload.eventType === 'UPDATE' || payload.eventType === 'INSERT') {

      setRecordings((prev: Recording[]) => [...prev.filter(x => x.id !== payload.old.id), {
        ...payload.new
      }])
    }
  }

  useEffect(() => {
    // listen to realtime inserts
    const realtime = supabase
      .channel('schema-db-changes')
      .on(
        'postgres_changes',
        {
          event: '*',
          schema: 'public',
          table: 'recordings',
        },
        onInsertOrUpdate)
      .subscribe()


    fetchRecordings()

    return () => {
      realtime.unsubscribe()
    }
  }, []);
```
  
  
  thanks anyway :/

**Reply by GaryAustin1** [reply]
Is the update for python being done many seconds after the subscription is set up?  There is a short time period after the subscription is connected before you can get changes.

**Reply by piccinnigius** [reply]
yes after about 1 minute
